languages/clojure/exercises/concept/basics/.docs/introduction.md

```diff
@@ -0,0 +1,42 @@
+In Clojure, binding a value to a name is referred to as a _var_. Top-level (global) vars are similar to constants in other languages, but are commonly redefined at a REPL to facilitate dynamic development.
+
+Top-level vars are defined using `def`:
+
+```clojure
+(def fingers 10)
+```
+
+The `defn` macro can be used to define a function taking zero or more arguments. A function always returns the result of the last expression in its body.
+
+```clojure
+(defn add [x y]
+  (+ x y))
+```
+
+Invoking a function is done by specifying its name and passing arguments for each of the function's parameters.
+
+```clojure
+(def five (add 2 3))
+```
+
+Functions and values in Clojure can only be used _after_ they have been defined. Using it before it has been defined results in a compile error.
+
+```clojure
+;; Compile error as `add` has not yet been defined
+(def seven (add 3 4))
```

Comment:
```suggestion
(def seven (add 3 4))

(defn add [x y]
  (+ x y))
```

Reply:
@ErikSchierboom this was a point of confusion for me on the first pass, since the `add` function was defined previously and the student is expected to "forget" that and consider the code blocks to be isolated.

In the F# exercise I forked this from, the line containing the `seven` binding is commented out. Is this to prevent an error should the code actually be executed by tooling or whatever? 

Perhaps it could be clarified, or I could also just be overthinking it.

Reply:
> In the F# exercise I forked this from, the line containing the seven binding is commented out. Is this to prevent an error should the code actually be executed by tooling or whatever?

Yes it is. I'm not sure yet what the best way to do this is. We've thought about being able to run the code snippets from the UI. In that case, it would make sense to uncomment the code to have it actually fail. For now, I wouldn't worry about this too much and just do whatever you prefer.